pytorch_lightning/loggers/neptune.py

```diff
@@ -46,235 +98,437 @@ class NeptuneLogger(LightningLoggerBase):
+            prefix: str = "training",
+            **neptune_run_kwargs):
+
+        # verify if user passed proper init arguments
+        self._verify_input_arguments(api_key, project, name, run, neptune_run_kwargs)
+
         super().__init__()
-        self.api_key = api_key
-        self.project_name = project_name
-        self.offline_mode = offline_mode
-        self.close_after_fit = close_after_fit
-        self.experiment_name = experiment_name
+        self._api_key = api_key
+        self._project = project
+        self._name = name
+        self._neptune_run_kwargs = neptune_run_kwargs
+        self._log_model_checkpoints = log_model_checkpoints
         self._prefix = prefix
-        self._kwargs = kwargs
-        self.experiment_id = experiment_id
-        self._experiment = None
 
-        log.info(f'NeptuneLogger will work in {"offline" if self.offline_mode else "online"} mode')
+        self._run_instance = run  # if run is None, instance will be initialized in first call to `run()`
+        self._run_instance_initialized = False
+
+    def _construct_path_with_prefix(self, *keys) -> str:
+        """Return sequence of keys joined by `LOGGER_JOIN_CHAR`, started with
+        `_prefix` if defined."""
+        if self._prefix:
+            return self.LOGGER_JOIN_CHAR.join([self._prefix, *keys])
+        return self.LOGGER_JOIN_CHAR.join(keys)
+
+    @staticmethod
+    def _verify_input_arguments(
+            api_key: Optional[str],
+            project: Optional[str],
+            name: Optional[str],
+            run: Optional["Run"],
+            neptune_run_kwargs: dict):
+
+        # check if user used legacy kwargs expected in `NeptuneLegacyLogger`
+        used_legacy_kwargs = [
+            legacy_kwarg for legacy_kwarg in neptune_run_kwargs
+            if legacy_kwarg in LEGACY_NEPTUNE_INIT_KWARGS
+        ]
+        if used_legacy_kwargs:
+            raise ValueError(
+                # TODO: product - review text
+                f"Following kwargs used by you are deprecated: {used_legacy_kwargs}.\n"
+                "If you are looking for the Neptune logger using legacy Python API it has been renamed to"
+                " NeptuneLegacyLogger. The NeptuneLogger was re-written to use the neptune.new Python API"
+                " (learn more: https://neptune.ai/blog/neptune-new).\n"
+                "You should use arguments accepted by either NeptuneLogger.init or neptune.init"
+            )
 
-    def __getstate__(self):
-        state = self.__dict__.copy()
+        # check if user used legacy kwargs expected in `NeptuneLogger` from neptune-pytorch-lightning package
+        used_legacy_neptune_kwargs = [
+            legacy_kwarg for legacy_kwarg in neptune_run_kwargs
+            if legacy_kwarg in LEGACY_NEPTUNE_LOGGER_KWARGS
+        ]
+        if used_legacy_neptune_kwargs:
+            raise ValueError(
+                # TODO: product - review text
```

Comment:
Any updates there ?

Reply:
Yes, @kamil-kaczmarek from our product team is currently working on updating docstrings and reviewing messages displayed to user.
The commit will be published today.